### PR TITLE
fix(azure): containerregistry_not_publicly_accesible is not accurate

### LINKED ISSUE #5937: AZURE check registry_not_publicly_accessible is working badly
### Steps to Reproduce

 python prowler.py azure --az-cli-auth    --subscription-id  XXXXXX --check registry_not_publicly_accessible

### Expected behavior

The check is indicating that it is FAIL when it is really Disabled. 
Line 23 , file containerregistry_not_publicly_accessible.py, it is being used network_rule_set property instead of public_network_access

### Actual Result with Screenshots or Logs

It does not generate a code error but rather a bad status result

### How did you install Prowler?

Cloning the repository from github.com (git clone)

### Environment Resource

WSL2 (Ubuntu)

### OS used

Windows

### Prowler version

Prowler 5.0.0 (You are running the latest version, yay!)

### Pip version

pip 24.0 from /home/xxxxxxx/.cache/pypoetry/virtualenvs/prowler-voSyvL3c-py3.12/lib/python3.12/site-packages/pip (python 3.12)

### Context

_No response_

### PR DESCRIPTION
**Pull request by StylusFrost**
### Context

Fixes #5937

### Description

The check is indicating that it is FAIL when it is really Disabled.

### Checklist

- Are there new checks included in this PR? No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/azure/services/containerregistry/containerregistry_service.py [hunk 1]
```diff
@@ -37,8 +37,10 @@ def _get_container_registries(self):
                                 resource_group=resource_group,
                                 sku=getattr(registry.sku, "name", ""),
                                 login_server=getattr(registry, "login_server", ""),
-                                public_network_access=getattr(
-                                    registry, "public_network_access", ""
+                                public_network_access=(
+                                    False
+                                    if registry.public_network_access == "Disabled"
```
Comment (puchy22) [inline prowler/providers/azure/services/containerregistry/containerregistry_service.py]:
```suggestion
                                    if getattr(registry, "public_network_access" "Enabled") == "Disabled"
```
Please add this, as according to [Microsoft documentation](https://learn.microsoft.com/en-us/python/api/azure-mgmt-containerregistry/azure.mgmt.containerregistry.v2023_11_01_preview.models.registry?view=azure-python) `public_network_access` is not mandatory and could be None.

File: tests/providers/azure/services/containerregistry/containerregistry_service_test.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def test_get_container_registry(self):
             assert registry_info.resource_group == "mock_resource_group"
             assert registry_info.sku == "Basic"
             assert registry_info.login_server == "mock_login_server.azurecr.io"
-            assert registry_info.public_network_access == "Enabled"
+            assert registry_info.public_network_access is False
```
Comment (puchy22) [inline tests/providers/azure/services/containerregistry/containerregistry_service_test.py]:
```suggestion
            assert not registry_info.public_network_access
```

**Review by puchy22 — CHANGES_REQUESTED** [review]
Great catch! Please review my comments and we can merge it.

**Comment by StylusFrost** [comment]
requested changes made 🚀

**Review by jfagoagas — APPROVED** [review]
Thanks for taking care of Prowler fixing this issue ❤️

**Review by puchy22 — APPROVED** [review]
Great catch! 💯

**Pull request merged**